File: src/1615.py
# https://chat.openai.com/share/d4307fac-a576-474e-bba3-43b2dae5fcdf

# Given array and k value
array = [1, 2, 3, 4, 5]
k = 2


# Function to rotate the array to the right by k elements in-place
def rotate_array(arr, k):
    n = len(arr)
    k = k % n  # In case k is larger than the array length

    # Reversing the entire array
    arr.reverse()

    # Reversing the first k elements
    arr[:k] = reversed(arr[:k])

    # Reversing the remaining elements
    arr[k:] = reversed(arr[k:])

    return arr


# Rotating the given array
rotated_array = rotate_array(array, k)
print(rotated_array)

File: src/1615.js
function rotateArray(arr, k) {
  // Ensure k is within the bounds of the array length
  k %= arr.length;

  // Helper function to reverse a portion of the array
  function reverse(start, end) {
    while (start < end) {
      let temp = arr[start];
      arr[start] = arr[end];
      arr[end] = temp;
      start++;
      end--;
    }
  }

  // Step 1: Reverse the entire array
  reverse(0, arr.length - 1);

  // Step 2: Reverse the first k elements
  reverse(0, k - 1);

  // Step 3: Reverse the remaining elements
  reverse(k, arr.length - 1);
}

// Example usage
let myArray = [1, 2, 3, 4, 5];
rotateArray(myArray, 2);
console.log(myArray); // Output will be [4, 5, 1, 2, 3]
